### PR TITLE
Sean temperature tracker

### PR DESCRIPTION
**Pull request by SeanGrady**

### CODE REVIEW

File: interview_cake/temperature_tracker/sean/temp_tracker.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        while running_total < self.num_inserted // 2:
+            running_total += self.count_dict[index]
+            index += 1
+        self.median = index
+
+    def _update_mode(self):
+        self.mode = max(
+            self.count_dict,
+            key = lambda x: self.count_dict[x],
+        )
```
Comment (ryanloughlin25) [inline interview_cake/temperature_tracker/sean/temp_tracker.py:49]: I think `max` on a dict like this is linear time.

**Review by ryanloughlin25 — CHANGES_REQUESTED** [review]
Requested Change:
Your function for finding the mode is not constant time.

Other comments:
Your tests directly check the properties of the temperature tracker object and the problem statement requested functions that return those values.  It's a small thing and I don't think you need to fix it.

File: interview_cake/temperature_tracker/sean/temp_tracker.py (hunk 1 above)
Comment (SeanGrady) [reply interview_cake/temperature_tracker/sean/temp_tracker.py:49]: It should be linear with the size of the dict, right? Which in this case is constant.

Comment (SeanGrady) [reply interview_cake/temperature_tracker/sean/temp_tracker.py:49]: I mean, it's not actually constant, but it's bounded at 111 things regardless of the number of temperatures you put in the object.

Comment (ryanloughlin25) [reply interview_cake/temperature_tracker/sean/temp_tracker.py:49]: okay you're right it is constant.  I forgot about the weirdness with constant input.  

Still, I think it's worth noting that you can do this without iterating through the dictionary.  Rather you could keep the previous mode temperature in a variable and only compare that count to the count of the temperature being inserted.

**Review by ryanloughlin25 — APPROVED** [review]

**Comment by ryanloughlin25** [comment]
testing github/slack integration

File: interview_cake/temperature_tracker/sean/temp_tracker.py (hunk 1 above)
Comment (SeanGrady) [reply interview_cake/temperature_tracker/sean/temp_tracker.py:49]: Oh, yeah, that's defs better

**Pull request merged**